src/NuGetPackageVerifier/Rules/AssemblyHasCommitHashAttributeRule.cs

```diff
@@ -11,18 +11,20 @@
 
 namespace NuGetPackageVerifier.Rules
 {
-    public class AssemblyHasCommitHashAttributeRule : AssemblyHasAttributeRuleBase
+    public class AssemblyHasCommitHashAttributeRule : IPackageVerifierRule
     {
-        public override IEnumerable<PackageVerifierIssue> ValidateAttribute(
-            string currentFilePath,
-            AssemblyDefinition assembly,
-            Collection<CustomAttribute> assemblyAttributes)
+        public IEnumerable<PackageVerifierIssue> Validate(PackageAnalysisContext context)
         {
-            var fileName = Path.GetFileNameWithoutExtension(currentFilePath);
-            var isSourcesPackage = fileName.EndsWith(".Sources", StringComparison.OrdinalIgnoreCase);
-            if (!isSourcesPackage && !HasCommitHashInMetadataAttribute(assemblyAttributes))
+            AssemblyHasAttributeHelper.GetAssemblyAttributesData(context);
```

Comment:
Hmm I agree that in practice it would be fine either way, but I kind of prefer it being on-demand because that better reflects the actual intentions of the code...